UI/DesignWnd.cpp

```diff
@@ -905,39 +905,41 @@ namespace {
     /** A class to allow the storage of the state of a GUI availabilty filter
         and the querying of that state WRT a ship design. */
     class AvailabilityManager {
-    private:
-        // A tuple of the toogle state of the 3-tuple of coupled
-        // availability filters in the GUI:
-        // Obsolete, Available and Unavailable
-        std::tuple<bool, bool, bool> m_availabilities;
-
     public:
+        // DisplayedAvailabiltyType is indexed by Availability::Enum
+        using DisplayedAvailabilty = std::tuple<bool, bool, bool>;
+
         AvailabilityManager(bool obsolete, bool available, bool unavailable);
 
+        const DisplayedAvailabilty& GetAvailabilities() const { return m_availabilities; };
         bool GetAvailability(const Availability::Enum type) const;
         void SetAvailability(const Availability::Enum type, const bool state);
         void ToggleAvailability(const Availability::Enum type);
 
         /** Given the GUI's displayed availabilities as stored in this
             AvailabiltyManager, return the displayed state of the design \p
             id. Return none if the design should not be displayed. */
-        boost::optional<Availability::Enum> DisplayedDesignAvailability(int id) const;
+        boost::optional<DisplayedAvailabilty> DisplayedDesignAvailability(int id) const;
         /** Given the GUI's displayed availabilities as stored in this
             AvailabiltyManager, return the displayed state of the hull \p
             name. Return none if the hull should not be displayed. */
-        boost::optional<Availability::Enum> DisplayedHullAvailability(const std::string& name) const;
+        boost::optional<DisplayedAvailabilty> DisplayedHullAvailability(const std::string& name) const;
         /** Given the GUI's displayed availabilities as stored in this
             AvailabiltyManager, return the displayed state of the part \p
             name. Return none if the part should not be displayed. */
-        boost::optional<Availability::Enum> DisplayedPartAvailability(const std::string& name) const;
+        boost::optional<DisplayedAvailabilty> DisplayedPartAvailability(const std::string& name) const;
 
     private:
         /** Given the GUI's displayed availabilities as stored in this
             AvailabiltyManager and that the X is \p available and \p obsolete,
             return the displayed state of the X. Return none if the X should
             not be displayed. */
-        boost::optional<Availability::Enum> DisplayedXAvailability(bool available, bool obsolete) const;
+        boost::optional<DisplayedAvailabilty> DisplayedXAvailability(bool available, bool obsolete) const;
 
+        // A tuple of the toogle state of the 3-tuple of coupled
+        // availability filters in the GUI:
+        // Obsolete, Available and Unavailable
+        std::tuple<bool, bool, bool> m_availabilities;
```

Comment:
probably use the typedef discussed above here...

Reply:
alternatively, make it an `map<Availability::Enum, bool>` ? would seem to simplify the data access without needing C++17 features.